emacs_lua_complete.lua

```diff
@@ -0,0 +1,81 @@
+--    __emacs_lua_complete:
```

Comment:
Nice step to move this away, it is clearly way too complicated to include in `lua-mode.el` file verbatim. One extra thing to do here is to figure out the packaging part. For example, we could put this into `scripts/` directory, and update MELPA recipe and the Cask file to include `(:files :defaults ("scripts", "*"))`.

Reply:
Yeah.  It also saves constant recompilation every time you complete, and opens the door for more "behind-the-scenes" interaction with lua going forward.  

The scripts sounds like a good plan. I created a `scripts` directory and referenced it.  I'm not really familiar with Cask; I think you have push access to this branch, if you don't mind sending a commit over.